lib/mayaUsd/ufe/UsdUndoCreateGroupCommand.cpp

```diff
@@ -79,24 +79,36 @@ void UsdUndoCreateGroupCommand::execute()
         setKindCmd->execute();
     }
 
-    auto newParentHierarchy = Ufe::Hierarchy::hierarchy(_group);
-    if (newParentHierarchy) {
-        for (auto child : _selection) {
-            auto parentCmd = newParentHierarchy->appendChildCmd(child);
-            parentCmd->execute();
-            append(parentCmd);
+    // Make sure to handle the exception if the parenting operation fails.
+    // This scenario happens if a user tries to group prim(s) in a layer
+    // other than the one where they were defined. In this case, the group creation itself
+    // will succeed, however the re-parenting is expected to throw an exception. We also need to
+    // make sure to undo the previous command ( AddNewPrimCommand ) when this happens.
+    try {
+        auto newParentHierarchy = Ufe::Hierarchy::hierarchy(_group);
+        if (newParentHierarchy) {
+            for (auto child : _selection) {
+                auto parentCmd = newParentHierarchy->appendChildCmd(child);
+                parentCmd->execute();
+                append(parentCmd);
+            }
         }
-    }
 
-    // Make sure to add the newly created _group (a.k.a parent) to selection. This matches native
-    // Maya behavior and also prevents the crash on grouping a prim twice.
-    Ufe::Selection groupSelect;
-    groupSelect.append(_group);
-    Ufe::GlobalSelection::get()->replaceWith(groupSelect);
+        // Make sure to add the newly created _group (a.k.a parent) to selection. This matches
+        // native Maya behavior and also prevents the crash on grouping a prim twice.
+        Ufe::Selection groupSelect;
+        groupSelect.append(_group);
+        Ufe::GlobalSelection::get()->replaceWith(groupSelect);
+
+        TF_VERIFY(
+            Ufe::GlobalSelection::get()->size() == 1,
+            "_group node should be in the global selection now. \n");
+    } catch (...) {
+        // undo previous AddNewPrimCommand
+        undo();
 
-    TF_VERIFY(
-        Ufe::GlobalSelection::get()->size() == 1,
-        "_group node should be in the global selection now. \n");
+        throw; // re-throw the same exception
+    }
 }
 
```

Comment:
@seando-adsk 
Grouping in a two tier operation:

1- create a group node
2- insert selected item(s) under the group node.

There could be a case where creating a group node would be Ok by the restriction rules but not the insertion part.
For example, this scenario could happen if a user tries to group prim(s) in a layer other than the one where they were defined. In this case, the group creation itself will succeed but not the insertion part. In this case, we should throw an exception and undo the previous group command. 

Also keep in mind that the way we are performing the insertion here makes it impossible to support grouping flags (e.g absolute, relative, etc...) . Eventually, the insertion logics needs to be moved in Maya so we can support those flags.